### DISCUSSION TITLE
Possible colored lasers improvement ideas

### DISCUSSION THREAD

**Discussion by QuadrupleAndy** [StrataSource/Portal-2-Community-Edition#805] [Ideas]
We came to a conclusion that the current state of the colored lasers is really bad for the new puzzle mechanic: it's more of a gimmick than a fulfilling testing element. Somewhere also lies the idea of a new cube type that can mix the colors of different lasers into one. (Not very friendly to color blind people though)
My personal suggestion is to make the lasers block each other making, as the lasers overall lack interaction with pretty much anything, so something like The Talos Principle.

So, we would appreciate hearing your thoughts on what should we do with it.

**Comment by ErinRoseWebs69** [comment]
i had opened an issue a while ago suggesting some ideas for improvements to the lasers, #789 . if you have any ideas for it, list em out and i can add them to that issue.

**Reply by QuadrupleAndy** [reply]
This discussion is about the campaign course and the puzzle mechanic behind it, not about the entity's customization.

**Reply by ErinRoseWebs69** [reply]
part of my issue suggests an ability to define custom behavior of the laser.

**Comment by ooooggll** [comment]
Not sure if this exists or has been suggested already, but colored laser catchers would be cool, so only a green laser can power a green catcher, red laser with red catcher, etc. Or colored cubes as well, where only a green cube can redirect a green laser, and a red laser will pass through/be stopped by a green cube.

**Reply by vrad-exe** [reply]
This already exists, but on its own it's not really interesting enough to make a bunch of puzzles out of.

**Reply by ooooggll** [reply]
I guess if that's not what they were asking, then having different colors have different functionalities could be interesting. Like have blue lasers fizzle cubes, green lasers do no contact damage, etc. Idk I don't have much ideas for individual mechanics like that but it's an idea.

**Reply by slambo311** [reply]
Different mechanics for different lasers is a much better idea. Heck, why not have something similar for the pellets?

**Reply by QuadrupleAndy** [reply]
That's probably will be like the countless custom fizzlers, confusing and not particularly interesting. The "no contact laser" is becoming the standard now in almost any good puzzle map in the workshop, as damaging the player never contributes to the puzzle solution, it's more of an annoyance than anything else. 
Speaking about the fizzling laser, IMO that would be very situational. You can do the same with the laser catchers connected to a dropper, buttons, and toxic goo. Although Rexaura uses the same concept for the pellets and cubes, it's not used as a standalone mechanic but as means to prevent pellets redirection on demand as it would break a lot of puzzles in the mod.
